MetaMask/metamask-extension
# Show a tx link and the "create a new swap" link for STX

## Explanation

This PR:
- Shows a tx link and the "create a new swap" link for STX on the STX status page
- Abstracts common code inside the Swaps feature for these 2 links

## Screenshots/Screencaps

STX enabled:
![image](https://user-images.githubusercontent.com/80175477/174804396-bb61be76-d175-4f2e-91ca-bdefa9a983da.png)

STX disabled (regular swap):
![image](https://user-images.githubusercontent.com/80175477/174799826-0e5a1248-1a8c-4197-9011-3d0e6dd5afa3.png)

## Manual Testing Steps
- Do a swap with STX (Smart Transactions) turned on. You can do that on the Build Quote page
- Once a smart transaction is done, you will see a transaction link and the "create a new swap" link on the STX status page
